atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockEntry.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.SortedMap;
+
+import org.apache.commons.lang3.StringUtils;
+import org.immutables.value.Value;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+
+@Value.Immutable
+public abstract class LockEntry {
```

Comment:
Also - the separator is needed if we want to have a lock ID, and a reason, and be able to CAS - CAS is only atomic for one cell, so we can only store one cell in this row (= one value).

I guess an alternative would be to store some string representation, e.g, `{"lockId"="-1","reason"="available"}`, although of course that would still need to be parsed.